src/Sentry.Unity/SentryUnity.cs

```diff
@@ -64,6 +65,18 @@ public static void Init(SentryUnityOptions options)
                             new ScreenshotAttachmentContent(options, SentryMonoBehaviour.Instance))));
                 }
 
+                if (options.NativeContextWriter is { } contextWriter)
+                {
+                    try
+                    {
+                        SentrySdk.ConfigureScopeAsync((scope) => Task.Run(() => contextWriter.Write(scope)));
+                    }
+                    catch (Exception e)
+                    {
+                        options.DiagnosticLogger?.LogWarning("Failed to synchronize scope to the native SDK: {0}", e);
```

Comment:
This catch block isn't going to be used since we're not awaiting nor blocking.

We should use Task.Run(() => ..).ContinuesWith(t => if t.IsSuccess, log error (observe t.Exception)